pkg/asset/installconfig/azure/validation.go

```diff
@@ -116,3 +118,44 @@ func validateRegion(client API, fieldPath *field.Path, p *aztypes.Platform) fiel
 
 	return field.ErrorList{field.Invalid(fieldPath, p.Region, fmt.Sprintf("region %q does not support resource creation", p.Region))}
 }
+
+// ValidateDNSRecordsDontExist checks DNS for CNAME, A, and AAA records for
+// api.zoneName. If a record exists, it's likely a cluster already exists.
+func ValidateDNSRecordsDontExist(ic *types.InstallConfig) error {
```

Comment:
nit: not a big fan of the name..

This function is actually checking if public DNS for cluster exists.. so something more appropriate would be better..

Reply:
I can't disagree, just couldn't think of anything more descriptive ;-) I've changed it to ValidatePublicDNS().